pdns/recursordist/stable-bloom.hh

```diff
@@ -106,15 +106,29 @@ public:
     uint8_t k, p;
     uint32_t num_cells, bitstr_len;
     is.read((char*)&k, sizeof(k));
+    if (is.fail()) {
+      throw std::runtime_error("SBF: read failed (file too short?)");
+    }
     is.read((char*)&num_cells, sizeof(num_cells));
+    if (is.fail()) {
+      throw std::runtime_error("SBF: read failed (file too short?)");
+    }
     num_cells = ntohl(num_cells);
     is.read((char*)&p, sizeof(p));
+    if (is.fail()) {
+      throw std::runtime_error("SBF: read failed (file too short?)");
+    }
     is.read((char*)&bitstr_len, sizeof(bitstr_len));
+    if (is.fail()) {
+      throw std::runtime_error("SBF: read failed (file too short?)");
+    }
     bitstr_len = ntohl(bitstr_len);
```

Comment:
I wonder if it would make sense to do some sanity check on that value before allocating the buffer?